In [1]:
import pandas as pd
import numpy as np
from matplotlib import pyplot as plt

In [2]:
store_data = pd.read_csv("./data/store.csv",index_col=0)

In [3]:
train_data = pd.read_csv("./data/train.csv")

/home/tjarke/anaconda3/envs/minichallenge/lib/python3.8/site-packages/IPython/core/interactiveshell.py:3145: DtypeWarning: Columns (7) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,


In [4]:
train_data["StateHoliday"].value_counts()

0      535644
0.0     63583
a       12798
b        4335
c        2160
Name: StateHoliday, dtype: int64

In [5]:
train_data.shape

(637774, 9)

In [6]:
store_data.shape

(1115, 9)

We do not have data for each store every day as you can see here:

In [9]:
train_data.shape[0] / store_data.shape[0]

571.9946188340807

In [10]:
train_data.isnull().any()

Date             False
Store             True
DayOfWeek         True
Sales             True
Customers         True
Open              True
Promo             True
StateHoliday      True
SchoolHoliday     True
dtype: bool

In [11]:
store_data.isnull().any()

StoreType                    False
Assortment                   False
CompetitionDistance           True
CompetitionOpenSinceMonth     True
CompetitionOpenSinceYear      True
Promo2                       False
Promo2SinceWeek               True
Promo2SinceYear               True
PromoInterval                 True
dtype: bool

In [15]:
for i in train_data:
    number_nan_rows = train_data.shape[0] - train_data.loc[~train_data.loc[:,i].isnull(),i].shape[0]
    percent_nan = number_nan_rows/train_data.shape[0]*100
    print(f"the nuber of nan rows for {i} is {number_nan_rows} and therefore nan is {round(percent_nan,2)} %")

the nuber of nan rows for Date is 0 and therefore nan is 0.0 %
the nuber of nan rows for Store is 19301 and therefore nan is 3.03 %
the nuber of nan rows for DayOfWeek is 19017 and therefore nan is 2.98 %
the nuber of nan rows for Sales is 19027 and therefore nan is 2.98 %
the nuber of nan rows for Customers is 19091 and therefore nan is 2.99 %
the nuber of nan rows for Open is 19186 and therefore nan is 3.01 %
the nuber of nan rows for Promo is 19194 and therefore nan is 3.01 %
the nuber of nan rows for StateHoliday is 19254 and therefore nan is 3.02 %
the nuber of nan rows for SchoolHoliday is 19337 and therefore nan is 3.03 %


In [16]:
for i in store_data:
    number_nan_rows = store_data.shape[0] - store_data.loc[~store_data.loc[:,i].isnull(),i].shape[0]
    percent_nan = number_nan_rows/store_data.shape[0]*100
    print(f"the nuber of nan rows for {i} is {number_nan_rows} and therefore nan is {round(percent_nan,2)} %")

the nuber of nan rows for StoreType is 0 and therefore nan is 0.0 %
the nuber of nan rows for Assortment is 0 and therefore nan is 0.0 %
the nuber of nan rows for CompetitionDistance is 3 and therefore nan is 0.27 %
the nuber of nan rows for CompetitionOpenSinceMonth is 354 and therefore nan is 31.75 %
the nuber of nan rows for CompetitionOpenSinceYear is 354 and therefore nan is 31.75 %
the nuber of nan rows for Promo2 is 0 and therefore nan is 0.0 %
the nuber of nan rows for Promo2SinceWeek is 544 and therefore nan is 48.79 %
the nuber of nan rows for Promo2SinceYear is 544 and therefore nan is 48.79 %
the nuber of nan rows for PromoInterval is 544 and therefore nan is 48.79 %


From the Nan values: For initial exploration drop columns :

CompetitionOpenSinceMonth is 354 and therefore nan is 31.75 %
CompetitionOpenSinceYear is 354 and therefore nan is 31.75 %



Promo2SinceWeek is 544 and therefore nan is 48.79 %
Promo2SinceYear is 544 and therefore nan is 48.79 %
PromoInterval is 544 and therefore nan is 48.79 %
